### PR TITLE
Add some French candidates for the display name

### PR DESCRIPTION
**Pull request by Gustry**
If you have other ideas @nirvn @DelazJ @rldhont @mdouchin

### CODE REVIEW

File: src/core/vector/qgsvectorlayerutils.cpp [hunk 1]
```diff
@@ -1200,15 +1200,22 @@ QString QgsVectorLayerUtils::guessFriendlyIdentifierField( const QgsFields &fiel
...
                                   QStringLiteral( "titel" ),  //#spellok
                                   QStringLiteral( "beschreibung" ),
                                   QStringLiteral( "strasse" ),
-                                  QStringLiteral( "beschriftung" ) };
+                                  QStringLiteral( "beschriftung" ),
+                                  // French candidates
+                                  QStringLiteral( "nom" ),
+                                  QStringLiteral( "titre" ),
+                                  QStringLiteral( "libelle" ),
+                                  QStringLiteral( "libellé" ),
```
Comment (m-kuhn) [inline src/core/vector/qgsvectorlayerutils.cpp:1215]: could we use `QString::normalized` instead?

File: src/core/vector/qgsvectorlayerutils.cpp [hunk 2]
```diff
@@ -1200,15 +1200,22 @@ QString QgsVectorLayerUtils::guessFriendlyIdentifierField( const QgsFields &fiel
                                   QStringLiteral( "title" ),
                                   QStringLiteral( "heibt" ),
```
Comment (uclaros) [inline src/core/vector/qgsvectorlayerutils.cpp:1201]: While here, you could move `heibt` under German candidates too.

Comment (m-kuhn) [reply src/core/vector/qgsvectorlayerutils.cpp:1201]: Is this german?

Comment (jef-n) [reply src/core/vector/qgsvectorlayerutils.cpp:1201]: > Is this german?

s/b/ß/

Comment (m-kuhn) [reply src/core/vector/qgsvectorlayerutils.cpp:1201]: > > Is this german?
> 
> s/b/ß/

Oh, the limited alphabet we Swiss live with

**Pull request closed without merging**